### PR TITLE
Add file upload and download SDK sample

### PR DESCRIPTION
**Pull request by weiyingmichellechen**
This sample shows how to upload and download a file over 16MB in size using SDK.

### CODE REVIEW

File: cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+using System.IO;
+
+namespace PowerApps.Samples
+{
+    public partial class SampleProgram
+    {
+        public static CrmServiceClient _service = null;
+        public static string entitySchemaName;
+        private static string columnSchemaName;
+        private const string entityId = "39a9c6f1-20dd-ec11-a7b6-0022482410a9";
```
Comment (phecke) [inline cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs:13]: You typically cannot hard code an ID. They are unique to an environment. You will need to do a Retrieve to get the ID unless you created the entity in the program and already have its ID.

The exception is that some templates and security roles have IDs that are fixed across environments and tenants.

File: cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Xrm.Sdk;
+using Microsoft.Xrm.Tooling.Connector;
+using System;
+using System.IO;
+
+namespace PowerApps.Samples
+{
+    public partial class SampleProgram
```
Comment (phecke) [inline cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs:8]: Add a ///<summary> comment of what the program does.

File: cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        [STAThread]
+        static void Main(string[] args)
+        {
+            Console.Write("Please enter Entity Logical Name (account): ");
+            entitySchemaName = Console.ReadLine();
+
+            Console.Write("Please enter column logical name (file1): ");
+            columnSchemaName = Console.ReadLine();
+
+            Console.Write("Please enter an uploaded file name, such as c:\\UploadedFiles\\document.pdf");
```
Comment (phecke) [inline cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs:25]: Probably want to ask for a file path to be consistent with your example.

File: cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                    var fileUploadDownload = new SampleFileUploadDownload(_service);
+                    var entity = new Entity(entitySchemaName, new Guid(entityId));
+                    var attributeName = publisherPrefix + "_" + columnSchemaName;
+
+                    fileUploadDownload.Upload(entity, uploadedFileName, Path.GetFileName(uploadedFileName), attributeName);
+
+                    var downloadFilePath = Path.Combine(Directory.GetCurrentDirectory());
+                    fileUploadDownload.Download(entity, downloadFilePath, attributeName);
+
+                    DeleteSolution(_service, solutionUniqueName);
```
Comment (phecke) [inline cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs:44]: Are you deleting the entity row you created earlier? Anything the program creates in Dataverse must be deleted before termination.

File: cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        private const string entityId = "39a9c6f1-20dd-ec11-a7b6-0022482410a9";
+        private static string uploadedFileName;
+
+        [STAThread]
+        static void Main(string[] args)
+        {
+            Console.Write("Please enter Entity Logical Name (account): ");
+            entitySchemaName = Console.ReadLine();
+
+            Console.Write("Please enter column logical name (file1): ");
```
Comment (phecke) [inline cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs:22]: Is this the File column logical name or the File column display name?

File: cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+            Console.Write("Please enter an uploaded file name, such as c:\\UploadedFiles\\document.pdf");
+            uploadedFileName = Console.ReadLine();
+
+            try
+            {
+                _service = SampleHelpers.Connect("Connect");
+                if (_service.IsReady)
+                {
+                    var publisherPrefix = CreatePublisher(_service);
```
[the remark this answers is not in the document]
Comment (JimDaly) [reply cds/orgsvc/C#/UploadDownloadFiles/SampleProgram.cs:33]: It is good not to use the default solution. It isn't recommended.

**Comment by phecke** [comment]
Closing this PR as we will not publish this code. We have already published other sample code with this functionality.

**Pull request closed without merging**